src/Sulu/Bundle/WebsiteBundle/Command/DumpSitemapCommand.php

```diff
@@ -0,0 +1,228 @@
+        }
+
+        $output->writeln('Start dumping "sitemap.xml" files:');
+
+        $webspaceManager = $this->getContainer()->get('sulu_core.webspace.webspace_manager');
+        foreach ($webspaceManager->getPortalInformations($this->environment) as $portalInformation) {
+            $this->dumpSitemap($portalInformation);
+        }
+    }
+
+    /**
+     * Dump sitemaps for portal-information.
+     *
+     * @param PortalInformation $portalInformation
+     */
+    private function dumpSitemap(PortalInformation $portalInformation)
+    {
+        if (-1 !== strpos($portalInformation->getUrl(), '{host}')) {
+            if (!$this->defaultHost) {
+                return;
+            }
+
+            $portalInformation->setUrl(str_replace('{host}', $this->defaultHost, $portalInformation->getUrl()));
+        }
+
+        if (!$this->pool->hasIndex()) {
+            return $this->dumpFile(
+                '/' . $portalInformation->getUrl() . '/sitemap.xml',
+                $this->renderProviderSitemap($this->pool->getFirstAlias(), $portalInformation)
+            );
+        }
+
+        $this->dumpFile('/' . $portalInformation->getUrl() . '/sitemap.xml', $this->renderIndex($portalInformation));
+
+        foreach ($this->pool->getProviders() as $alias => $provider) {
+            $this->dumpFile(
+                '/' . $portalInformation->getUrl() . '/sitemaps/' . $alias . '.xml',
+                $this->renderProviderSitemap($alias, $portalInformation)
+            );
+        }
+    }
+
+    /**
+     * Render sitemap for provider.
+     *
+     * @param string $alias
+     * @param PortalInformation $portalInformation
+     *
+     * @return string
+     */
+    private function renderProviderSitemap($alias, PortalInformation $portalInformation)
+    {
+        $provider = $this->pool->getProvider($alias);
+        if (1 >= ($maxPage = (int) $provider->getMaxPage())) {
+            return $this->renderSitemap($alias, 1, $portalInformation);
+        }
+
+        $pathFormat = '/%s/sitemaps/%s-%s.xml';
+        for ($page = 1; $page <= $maxPage; ++$page) {
+            $path = sprintf($pathFormat, $portalInformation->getUrl(), $alias, $page);
+            $this->dumpFile($path, $this->renderSitemap($alias, $page, $portalInformation));
+        }
+
+        return $this->render(
+            'SuluWebsiteBundle:Sitemap:sitemap-paginated-index.xml.twig',
+            ['alias' => $alias, 'maxPage' => $maxPage]
+        );
+    }
+
+    /**
+     * Render sitemap.
+     *
+     * @param string $alias
+     * @param int $page
+     * @param PortalInformation $portalInformation
+     *
+     * @return string
+     */
+    private function renderSitemap($alias, $page, PortalInformation $portalInformation)
+    {
+        $provider = $this->pool->getProvider($alias);
+        $entries = $provider->generate($page, $portalInformation->getPortalKey(), $portalInformation->getLocale());
```

Comment:
Generate sounds to me a bit like this method creates new data. Would maybe `build` be a better name?